server/etcdserver/api/v3rpc/interceptor.go

```diff
@@ -76,13 +76,59 @@ func newLogUnaryInterceptor(s *etcdserver.EtcdServer) grpc.UnaryServerIntercepto
 		startTime := time.Now()
 		resp, err := handler(ctx, req)
 		lg := s.Logger()
-		if lg != nil { // acquire stats if debug level is enabled or request is expensive
+		if lg != nil { // acquire stats if debug level is enabled or RequestInfo is expensive
 			defer logUnaryRequestStats(ctx, lg, s.Cfg.WarningUnaryRequestDuration, info, startTime, req, resp)
 		}
 		return resp, err
 	}
 }
 
+type GrpcRecorder struct {
```

Comment:
Can we move it to `pkg/grpc_testing` ?

I assume it should be only used in the tests.

Reply:
Done